File: orders/src/main/java/com/github/orders/dto/OrderDetailEntryDto.java
package com.github.orders.dto;

import com.github.orders.entity.OrderStatus;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class OrderDetailEntryDto {

    private Long id;

    private CustomerDto customer;

    private List<ProductDetailDto> productIds;

    private BigDecimal amount;

    private Long userId;

    private OrderStatus status;

}

File: server-mocks/src/main/kotlin/com/github/server/mocks/payload/OrderDetailEntryDto.kt
package com.github.server.mocks.payload

import com.fasterxml.jackson.annotation.JsonProperty
import lombok.AllArgsConstructor
import lombok.Data
import lombok.NoArgsConstructor
import java.math.BigDecimal

@Data
@AllArgsConstructor
@NoArgsConstructor
class OrderDetailEntryDto {
    @JsonProperty(value = "id")
    private val id: Long? = null

    @JsonProperty(value = "customer")
    private val customer: CustomerDto? = null

    @JsonProperty(value = "productIds")
    private val productIds: List<ProductDto>? = null

    @JsonProperty(value = "amount")
    private val amount: BigDecimal? = null

    @JsonProperty(value = "userId")
    private val userId: Long? = null

    @JsonProperty(value = "status")
    private val status: OrderStatus? = null
}